docs/en/pact-reference.md

```diff
@@ -2671,17 +2671,19 @@ They are used in [with-read](pact-functions.html#with-read), [with-default-read]
 ```
 
 ### Lambdas {#lambdas}
-Lambdas are supported in `let` and `let*`.
-Note: Lambdas are not atoms, since in pact we do not allow arbitrary recursion, therefore
-they must be let-bound and named.
+Lambdas are supported in `let`, `let*`, and as inline arguments to built-in function applications.
 
 ```lisp
   ; identity function
   (let ((f (lambda (x) x))) (f a))
-  ; Higher order example
-  (let ((hof (lambda (f a b)  (f a b))))  (hof + 1 2))
+  ; native example
+  (let ((f (lambda (x) x))) (map (f) [1 2 3]))
+  ; Inline native example:
+  (map (lambda (x) x) [1 2 3])
 ```
 
+Note: Lambdas and defs as function arguments are currently not supported outside of built-in functions: `map`, `zip`, `filter`, `fold`, `fold-db`, `and?`, `or?`, `not?`, `compose`.
```

Comment:
I wouldn't add this line: it's a maintenance burden to maintain the list, plus it doesn't really implicate lambdas per se. If you want to update https://github.com/kadena-io/pact/blob/95c5c9016d73ad2f9d355c10228e391f8d3b964b/docs/en/pact-reference.md#partial-application-partialapplication that's ok (and fix the broken links elsewhere to #partial-application :) )